### DISCUSSION TITLE
How to align on both side of a panel and leave flexible space in between

### DISCUSSION THREAD

**Discussion by WarWithinMe** [AvaloniaUI/Avalonia#11475] [Q&A] [answered]
In the screenshot below, I want to push the red button to the furthest right:

![屏幕截图 2023-05-22 163334](https://github.com/AvaloniaUI/Avalonia/assets/567110/e304a907-2e64-411a-808f-5967026550e4)

What I've tried is to use a DockPanel, and set the Red Button Dock to the Right. But somehow it doesn't work. Does anybody have any idea?
```
<StackPanel Background="White" Orientation="Vertical" HorizontalAlignment="Stretch">
    <StackPanel Height="32" Orientation="Horizontal">
        <Button>
            <PathIcon Data="..." />
        </Button>
        <Label>Demo</Label>
    </StackPanel>

    <DockPanel Height="32" HorizontalAlignment="Stretch">
        <StackPanel Orientation="Horizontal" HorizontalAlignment="Stretch">
            <Button>
                <PathIcon Foreground="Blue" Data="..." />
            </Button>
            <Button>
                <PathIcon Foreground="Blue" Data="..." />
            </Button>
            <Button>
                <PathIcon Foreground="Blue" Data="..." />
            </Button>
        </StackPanel>

        <Button DockPanel.Dock="Right">
            <PathIcon Foreground="Red" Data="..." />
        </Button>
    </DockPanel>
</StackPanel>
```

The desired effect should look like this:
![image](https://github.com/AvaloniaUI/Avalonia/assets/567110/6ef95a95-c574-48b0-a3a8-8eda6d9ca6b1)

**Comment by MarkusKgit** [comment, marked as answer]
By default the last item of a DockPanel takes up the remaining space. You can either set `LastChildFill="False"` on the DockPanel or move the Button above the StackPanel
